### PR TITLE
fix: notification api support check

### PR DESCRIPTION
**Pull request by hdz-666**
### Details(context, link the issue, how was the bug fixed, what does the new feature do)

-
-

### CODE REVIEW

File: packages/react-sdk/src/hooks/useAwayNotifications.ts [hunk 1]
```diff
@@ -7,8 +7,8 @@ import { useHMSVanillaStore } from '../primitives/HmsRoomProvider';
 export const useAwayNotifications = () => {
   const vanillaStore = useHMSVanillaStore();
   const requestPermission = useCallback(async () => {
-    // Headless check for beam
-    if (navigator.webdriver) {
+    if (!('Notification' in window) || navigator.webdriver) {
+      // Notifications not supported
```
Comment (KaustubhKumar05) [inline packages/react-sdk/src/hooks/useAwayNotifications.ts:11]: Let's log this too?

**Review by KaustubhKumar05 — APPROVED** [review]

File: packages/react-sdk/src/hooks/useAwayNotifications.ts [hunk 2]
```diff
@@ -23,6 +25,11 @@ export const useAwayNotifications = () => {
   }, [vanillaStore]);
 
   const showNotification = useCallback((title: string, options?: NotificationOptions) => {
+    // Notifications not supported
+    if (!('Notification' in window)) {
+      console.debug('Show Notifications: Notifications not supported or headless browser');
+      return;
+    }
     if (
       !Notification ||
```
Comment (raviteja83) [inline packages/react-sdk/src/hooks/useAwayNotifications.ts:34]: !Notification seems irrelevant now I guess. can we rmeove?

**Review by ygit — APPROVED** [review]
LGTM

File: packages/react-sdk/src/hooks/useAwayNotifications.ts (hunk 2 above)
Comment (KaustubhKumar05) [reply packages/react-sdk/src/hooks/useAwayNotifications.ts:34]: @hdz-666 can you resolve this?

**Review by raviteja83 — APPROVED** [review]

File: packages/react-sdk/src/hooks/useAwayNotifications.ts [hunk 3]
```diff
@@ -23,8 +25,12 @@ export const useAwayNotifications = () => {
   }, [vanillaStore]);
 
   const showNotification = useCallback((title: string, options?: NotificationOptions) => {
+    // Notifications not supported
+    if (!('Notification' in window)) {
+      console.debug('Show Notifications: Notifications not supported or headless browser');
+      return;
+    }
```
Comment (amar-1995) [inline packages/react-sdk/src/hooks/useAwayNotifications.ts:32]: Why check here again?

Comment (raviteja83) [reply packages/react-sdk/src/hooks/useAwayNotifications.ts:32]: to not error when showNotification is used without requesting permissions. edge case

**Review by amar-1995 — APPROVED** [review]
